### PR TITLE
Bug/ep 358 cpu usage of b one gateway too high subscription fix

### PR DESCRIPTION
**Pull request by MaHaBa22**
# PR Checklist

> **If your build fails** due to your commit message not passing the build checks, please review the guidelines [here](https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md).

Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?)

## Testing Instructions

1. Connect to opcua device with S7 using no security( no username, no password, Policy= None, Mode= None)
   - Subscriptions will start ( see device opcua container log
   - disconnect the machine -> subscriptions stop
   - reconnect machine-> subscriptions should start again
2. Connect to opcua device with Simotion using security(username, password, Policy = Basic256Sha256, Mode= SignAndEncrypt)
   - Subscriptions will start ( see device opcua container log
   - disconnect the machine -> subscriptions stop
   - reconnect machine-> subscriptions should start again

### CODE REVIEW

**Comment by JPLuehrig** [comment]
maybe use at the end of the Initialize the:
     go d.startSubscriber()

Not as easy as expected, non blocking for loop to start subcription initially

File: internal/driver/subscriptionlistener.go [hunk 1]
```diff
@@ -49,64 +51,131 @@ func (d *Driver) startSubscriptionListener() error {
...
 		case <-ctx.Done():
 			return nil
 			// receive Publish Notification Data
-		case res := <-sub.Notifs:
+		case res := <-notifyCh:
 			if res.Error != nil {
 				d.Logger.Debug(res.Error.Error())
 				continue
 			}
 			switch x := res.Value.(type) {
```
Comment (mchavezN) [inline internal/driver/subscriptionlistener.go]: I will suggest renaming the "x" to notification or changeNotification:` switch changeNotification :=` makes it more readable.

File: internal/driver/subscriptionlistener.go [hunk 2]
```diff
@@ -49,64 +51,131 @@ func (d *Driver) startSubscriptionListener() error {
...
+	Close() error
+}
+
+// SubscriptionCanceller interface gives us possibility to mock opcua client functions for cling a client in tests.
+type SubscriptionCanceller interface {
+	Cancel(ctx context.Context) error
+}
+
+// CloseClient tries to close the client connection
+func CloseClient(d *Driver, client ClientCloser) error {
```
Comment (mchavezN) [inline internal/driver/subscriptionlistener.go]: What about renaming CloseClientConnection? It is what this function does

**Review by mchavezN — CHANGES_REQUESTED** [review]

File: internal/driver/subscriptionlistener.go (hunk 1 above)
Comment (MaHaBa22) [reply internal/driver/subscriptionlistener.go]: makes sense to me

File: internal/driver/subscriptionlistener.go (hunk 2 above)
Comment (MaHaBa22) [reply internal/driver/subscriptionlistener.go]: done

**Review by mchavezN — APPROVED** [review]
LGTM

**Pull request merged**